manager/manager.go

```diff
@@ -95,14 +95,85 @@ type MapSpecEditor struct {
 	EditorFlag MapSpecEditorFlag
 }
 
+// ProbesSelector - A probe selector defines how a probe (or a group of probes) should be activated.
+//
+// For example, this can be used to specify that out of a group of optional probes, at least one should be activated.
+type ProbesSelector interface {
+	// GetProbesList - Returns the list of probes that this selector activates
+	GetProbesList() []ProbeSelector
+	// RunValidator - Ensures that the probes that were successfully activated follow the selector goal.
+	// For example, see OneOf.
+	RunValidator(manager *Manager) error
+	// String - Stringifies the probe selector
+	String() string
+}
+
+// ProbeSelector - This selector is used to unconditionally select a probe by its identification pair
+type ProbeSelector struct {
+	ProbeIdentificationPair
+}
+
+func (ps ProbeSelector) GetProbesList() []ProbeSelector {
+	return []ProbeSelector{ps}
+}
+
+func (ps ProbeSelector) RunValidator(manager *Manager) error {
+	p, ok := manager.GetProbe(ps.ProbeIdentificationPair)
+	if !ok {
+		return errors.Errorf("probe not found: %s", ps.ProbeIdentificationPair)
+	}
+	if !p.IsRunning() && p.Enabled {
+		return errors.Errorf("probe %s is not running, however it is enabled", ps.ProbeIdentificationPair)
+	}
+	return nil
+}
+
+// OneOf - This selector is used to ensure that at least of a list of probe selectors is valid. In other words, this
+// can be used to ensure that at least one of a list of optional probes is activated.
+type OneOf struct {
+	Selectors []ProbesSelector
+}
+
+func (oo OneOf) GetProbesList() []ProbeSelector {
+	var l []ProbeSelector
+	for _, selector := range oo.Selectors {
+		l = append(l, selector.GetProbesList()...)
+	}
+	return l
+}
+
+func (oo OneOf) RunValidator(manager *Manager) error {
+	var valid bool
+	for _, selector := range oo.Selectors {
+		if err := selector.RunValidator(manager); err == nil {
+			valid = true
```

Comment:
nit: you could `return nil` here and get rid of the `valid` variable.

Reply:
👍